src/Microsoft.AspNet.Identity.EntityFramework/IdentityDbContext.cs

```diff
@@ -48,62 +48,59 @@ protected IdentityDbContext()
+                b.Property(u => u.NormalizedUserName).HasMaxLength(256);
+                b.Property(u => u.Email).HasMaxLength(256);
+                b.Property(u => u.NormalizedEmail).HasMaxLength(256);
+                b.HasMany(u => u.Claims).WithOne().HasForeignKey(uc => uc.UserId).OnDelete(DeleteBehavior.Cascade);
+                b.HasMany(u => u.Logins).WithOne().HasForeignKey(ul => ul.UserId).OnDelete(DeleteBehavior.Cascade);
+                b.HasMany(u => u.Roles).WithOne().HasForeignKey(ur => ur.UserId).OnDelete(DeleteBehavior.Cascade);
+            });
 
             builder.Entity<TRole>(b =>
-                {
-                    b.HasKey(r => r.Id);
-                    b.HasIndex(r => r.NormalizedName).HasName("RoleNameIndex");
-                    b.ToTable("AspNetRoles");
-                    b.Property(r => r.ConcurrencyStamp).IsConcurrencyToken();
+            {
+                b.HasKey(r => r.Id);
+                b.HasIndex(r => r.NormalizedName).HasName("RoleNameIndex");
+                b.ToTable("AspNetRoles");
+                b.Property(r => r.ConcurrencyStamp).IsConcurrencyToken();
 
-                    b.Property(u => u.Name).HasMaxLength(256);
-                    b.Property(u => u.NormalizedName).HasMaxLength(256);
+                b.Property(u => u.Name).HasMaxLength(256);
+                b.Property(u => u.NormalizedName).HasMaxLength(256);
 
-                    b.HasMany(r => r.Users).WithOne().HasForeignKey(ur => ur.RoleId);
-                    b.HasMany(r => r.Claims).WithOne().HasForeignKey(rc => rc.RoleId);
-                });
+                b.HasMany(r => r.Users).WithOne().HasForeignKey(ur => ur.RoleId).OnDelete(DeleteBehavior.Cascade);
+                b.HasMany(r => r.Claims).WithOne().HasForeignKey(rc => rc.RoleId).OnDelete(DeleteBehavior.Cascade);
+            });
 
             builder.Entity<IdentityUserClaim<TKey>>(b =>
-                {
-                    b.HasKey(uc => uc.Id);
-                    b.ToTable("AspNetUserClaims");
-                });
+            {
+                b.HasKey(uc => uc.Id);
+                b.ToTable("AspNetUserClaims");
+            });
 
             builder.Entity<IdentityRoleClaim<TKey>>(b =>
-                {
-                    b.HasKey(rc => rc.Id);
-                    b.ToTable("AspNetRoleClaims");
-                });
+            {
+                b.HasKey(rc => rc.Id);
+                b.ToTable("AspNetRoleClaims");
+            });
 
             builder.Entity<IdentityUserRole<TKey>>(b =>
-                {
-                    b.HasKey(r => new { r.UserId, r.RoleId });
-                    b.ToTable("AspNetUserRoles");
-                });
-            // Blocks delete currently without cascade
-            //.ForeignKeys(fk => fk.ForeignKey<TUser>(f => f.UserId))
-            //.ForeignKeys(fk => fk.ForeignKey<TRole>(f => f.RoleId));
+            {
+                b.HasKey(r => new { r.UserId, r.RoleId });
```

Comment:
I want all subentities to be deleted if the root object goes away (user => roles/claims/logins role => roles)